src/main/java/org/opentripplanner/routing/algorithm/filterchain/filters/system/OutsideSearchWindowFilter.java

```diff
@@ -35,8 +41,21 @@ public String name() {
   @Override
   public Predicate<Itinerary> shouldBeFlaggedForRemoval() {
     return it -> {
-      var time = it.startTime().toInstant();
-      return time.isBefore(earliestDepartureTime) || !time.isBefore(latestDepartureTime);
+      var startTime = it.startTime().toInstant();
+      // for arrive by searches the street/flex-only results are treated differently:
+      // arrive-by transit result are filtered by their departure time and whether they don't depart
+      // after the end of the computed search window which is dependent on the heuristic's minimum
+      // transit time.
```

Comment:
This is slightly difficult to understand. Maybe use bullets for listing the criteria for filtering. Also, I'm a bit on the edge if this should belong here or on the javadoc for this method as this method doesn't really contain anything else.

Reply:
I converted it to Javadoc and tried to make it more understandable.